### PR TITLE
Continue removing code relating to Policy pages

### PR DESCRIPTION
**Pull request by cbaines**
This commit removes the bulk of the code related to Policy pages,
particularly related to tagging, models and sending content to the
Publishing API.

### CODE REVIEW

**Comment by sihugh** [comment]
@cbaines this looks cool, but there's a bunch of conflicts now 😞

**Comment by cbaines** [comment]
> @cbaines this looks cool, but there's a bunch of conflicts now disappointed

Yeah, I might try and make some time over the next week to rebase this...

**Comment by cbaines** [comment]
@kevindew I think I remember that quite a while ago, you volunteered to maybe review this?

If you're still interested, I've now fixed the merge conflicts. I'm also unsure whether this is best as one commit, or whether it's worth splitting it up in to multiple commits.

**Comment by cbaines** [comment]
I've rebased again to fix the merge issues.

**Comment by kevindew** [comment]
> I've rebased again to fix the merge issues.

Great - btw I'm up to 42 of 131 files reviewed so far. Hopefully will get through it all soon.

**Comment by cbaines** [comment]
> Looks great - awesome job Chris.

Great :) Thanks for taking a look.

> Only thing I found a bit surprising was the changed tests in: https://github.com/alphagov/whitehall/pull/4450/files#diff-f1b0ef9a25f7b9e50da72ff68bbd9021R226-R230 and https://github.com/alphagov/whitehall/pull/4450/files#diff-300c2f297a5b0f38f4c5574ddbb880b6R38-R42 I assume they're because we've had to generalise specific tests but thought I'd flag it in case these surprise you at all.

Which files/tests are changed? The links don't seem to identify them for me, I just end up at the top of the diff. Perhaps put some comments on the diff itself.

File: features/step_definitions/publication_steps.rb [hunk 1]
```diff
@@ -35,13 +35,10 @@
...
-  click_button "Save and continue"
-  click_button "Save and review legacy tagging"
-  select first_policy, from: "Policies"
-  select second_policy, from: "Policies"
-  click_button "Save"
+When(/^I visit the list of publications$/) do
+  stub_content_item_from_content_store_for(publications_path)
+  visit homepage
+  click_link "Publications"
 end
```
Comment (kevindew) [inline features/step_definitions/publication_steps.rb]: @cbaines this test

File: features/step_definitions/document_steps.rb [hunk 2]
```diff
@@ -223,11 +223,11 @@
...
-Then(/^the publication should be related to "([^"]*)" and "([^"]*)" policies$/) do |related_policy_1, related_policy_2|
-  policies_titles = Publication.last.policies.map(&:title)
-
-  assert policies_titles.include?(related_policy_1)
-  assert policies_titles.include?(related_policy_2)
+Then(/^(#{THE_DOCUMENT}) should be visible to the public$/) do |edition|
+  public_edition_path = public_path_for(edition)
+  stub_content_item_from_content_store_for(public_edition_path)
+  visit_public_index_for(edition)
+  assert page.has_css?(record_css_selector(edition), text: edition.title)
```
Comment (kevindew) [inline features/step_definitions/document_steps.rb]: and this test

File: features/step_definitions/publication_steps.rb (hunk 1 above)
Comment (cbaines) [reply features/step_definitions/publication_steps.rb]: Good spot, I think this change is a result of a bad rebase. This step definition was removed in 79e3bf6eb4bac18c40e186f5edd3d718bb0e969e.

File: features/step_definitions/document_steps.rb (hunk 2 above)
Comment (cbaines) [reply features/step_definitions/document_steps.rb]: This was also removed in 79e3bf6eb4bac18c40e186f5edd3d718bb0e969e, and came back from a bad rebase. I'll fix this as well.

**Pull request merged**